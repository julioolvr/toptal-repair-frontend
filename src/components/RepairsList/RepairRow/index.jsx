import React from 'react';
import PropTypes from 'prop-types';
import g from 'glamorous';

import repairPropType from '../../../proptypes/repair';

function RepairRow({ repair, canDeleteRepair, onDeleteRepair }) {
  return (
    <g.Div padding=".8em" borderBottom="1px solid rgba(0, 0, 0, .1)">
      {repair.title}
      {canDeleteRepair && (
        <button
          onClick={(e) => {
            e.preventDefault();
            onDeleteRepair(repair.id);
          }}
        >
          X
        </button>
      )}
    </g.Div>
  );
}

RepairRow.propTypes = {
  repair: repairPropType.isRequired,
  canDeleteRepair: PropTypes.bool.isRequired,
  onDeleteRepair: PropTypes.func,
};

RepairRow.defaultProps = {
  onDeleteRepair: () => {},
};

export default RepairRow;
