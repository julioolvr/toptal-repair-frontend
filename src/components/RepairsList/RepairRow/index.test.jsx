import React from 'react';
import { shallow } from 'enzyme';

import RepairRow from '.';

it('renders a row for a repair', () => {
  const wrapper = shallow(<RepairRow repair={{ title: 'Test repair' }} canDeleteRepair={false} />);
  expect(wrapper).toMatchSnapshot();
});

it('renders a row for a repair that can be deleted', () => {
  const wrapper = shallow(<RepairRow repair={{ title: 'Test repair' }} canDeleteRepair />);
  expect(wrapper).toMatchSnapshot();
});

it('calls onDeleteRepair with the repair id when clicking on the X button', () => {
  const repair = { id: 42 };
  const onDeleteRepair = jest.fn();

  const wrapper = shallow(
    <RepairRow repair={repair} canDeleteRepair onDeleteRepair={onDeleteRepair} />,
  );

  wrapper
    .findWhere(el => el.is('button') && el.text() === 'X')
    .simulate('click', { preventDefault: () => {} });

  expect(onDeleteRepair).toHaveBeenCalledWith(repair.id);
});
