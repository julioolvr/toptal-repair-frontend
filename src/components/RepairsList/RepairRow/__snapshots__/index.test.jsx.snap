// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a row for a repair 1`] = `
<glamorous.Div
  borderBottom="1px solid rgba(0, 0, 0, .1)"
  padding=".8em"
>
  Test repair
</glamorous.Div>
`;

exports[`renders a row for a repair that can be deleted 1`] = `
<glamorous.Div
  borderBottom="1px solid rgba(0, 0, 0, .1)"
  padding=".8em"
>
  Test repair
  <button
    onClick={[Function]}
  >
    X
  </button>
</glamorous.Div>
`;
